### PR TITLE
MPI4: add support for mpi_info_get_string

### PR DESCRIPTION
**Pull request by hppritcha**
related to #9192

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

**Comment by hppritcha** [comment]
looks like MLNX CI is having a really bad docker day:
```
Starting: Initialize containers
/usr/bin/docker version --format '{{.Server.APIVersion}}'
'1.40'
Docker daemon API version: '1.40'
/usr/bin/docker version --format '{{.Client.APIVersion}}'
'1.40'
Docker client API version: '1.40'
/usr/bin/docker ps --all --quiet --no-trunc --filter "label=212c59"
/usr/bin/docker network prune --force --filter "label=212c59"
/usr/bin/docker pull rdmz-harbor.rdmz.labs.mlnx/hpcx/ompi_ci:latest
Error response from daemon: Get http://rdmz-harbor.rdmz.labs.mlnx/v2/: dial tcp 10.224.1.1:80: connect: connection refused
##[warning]Docker pull failed with exit code 1, back off 1.055 seconds before retry.
/usr/bin/docker pull rdmz-harbor.rdmz.labs.mlnx/hpcx/ompi_ci:latest
Error response from daemon: Get http://rdmz-harbor.rdmz.labs.mlnx/v2/: dial tcp 10.224.1.1:80: connect: connection refused
##[warning]Docker pull failed with exit code 1, back off 8.358 seconds before retry.
/usr/bin/docker pull rdmz-harbor.rdmz.labs.mlnx/hpcx/ompi_ci:latest
Error response from daemon: Get http://rdmz-harbor.rdmz.labs.mlnx/v2/: dial tcp 10.224.1.1:80: connect: connection refused
##[error]Docker pull failed with exit code 1
Finishing: Initialize containers
```

File: ompi/mpi/fortran/mpif-h/info_get_string_f.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                if (OMPI_SUCCESS !=
+                  (ret = ompi_fortran_string_c2f(info_str->string, value, value_len + 1))) {
+                    c_ierr = OMPI_ERRHANDLER_NOHANDLE_INVOKE(ret, FUNC_NAME);
+                    if (NULL != ierr) {
+                        *ierr = OMPI_INT_2_FINT(c_ierr);
+                    }
+                }
+                *buflen = info_str->length;
+            }
+            OBJ_RELEASE(info_str);
```
Comment (devreal) [inline ompi/mpi/fortran/mpif-h/info_get_string_f.c:136]: I think this line should move into the block above

**Review by devreal — CHANGES_REQUESTED** [review]

File: ompi/mpi/fortran/mpif-h/info_get_string_f.c (hunk 1 above)
Comment (hppritcha) [reply ompi/mpi/fortran/mpif-h/info_get_string_f.c:136]: actually I think you mean below.  but its not in the right place current.

Comment (hppritcha) [reply ompi/mpi/fortran/mpif-h/info_get_string_f.c]: @devreal you're right, pushed fix.

**Review by devreal — APPROVED** [review]

**Comment by awlauria** [comment]
if this is good to go please merge and cherry-pick to v5.0.x @gpaulsen @janjust

**Comment by hppritcha** [comment]
@awlauria waiting on other reviewers....

**Comment by hppritcha** [comment]
hmm... actually should add a man page to this pr too.

**Pull request merged**